apps/kubernetes-express-otel/express-otel-app/app.yaml

```
@@ -0,0 +1,43 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: app
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: app
+  template:
+    metadata:
+      labels:
+        app: app
+    spec:
+      containers:
+      - name: app
+        image: otel-js-app:1.0
+        imagePullPolicy: Never
+        ports:
+        - containerPort: 3000
+        env:
+         - name: HOST_IP
+           valueFrom:
+             fieldRef:
+               fieldPath: status.hostIP
+         - name: OTEL_EXPORTER_OTLP_ENDPOINT
+           value: "http://$(HOST_IP):4318/v1/traces"
```

Per [the spec](https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/#otel_exporter_otlp_endpoint), `OTEL_EXPORTER_OTLP_ENDPOINT` is 
> A base endpoint URL **for any signal type**, with an optionally-specified port number. Helpful for when you’re sending more than one signal to the same endpoint and want one environment variable to control the endpoint.

but here you are passing the traces endpoint.

To fix this we can either pass the generic endpoint:
```suggestion
         - name: OTEL_EXPORTER_OTLP_ENDPOINT
           value: "http://$(HOST_IP):4318"
```

or use the traces-specific environment variable:
```suggestion
         - name: OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
           value: "http://$(HOST_IP):4318/v1/traces"
```

---

Oh! I see the difference now, when I read https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/ .